arviz/plots/distcomparisonplot.py

```diff
@@ -133,6 +134,14 @@ def plot_dist_comparison(
         list(xarray_var_iter(data, var_names=var, combined=True))[:len_plots]
         for data, var in zip(datasets, var_names)
     ]
+    if len(dc_plotters) + 1 > len_plots:
+        warnings.warn(
+            "rcParams['plot.max_subplots'] ({max_plots}) is smaller than the number "
+            "of variables to plot ({len_plotters}), generating only {max_plots} "
+            "plots".format(max_plots=len_plots, len_plotters=len(dc_plotters)),
+            UserWarning,
+        )
+        dc_plotters = dc_plotters[:len_plots]
```

Comment:
the `xarray_var_iter` list is still getting truncated when defining `dc_plotters`, and it should be truncated at that level but afterwards while printing the warning. 

`dc_plotters` will have length of 2 or 3 always, as it is a list of lists, it's those sublists the ones that need to be truncated.

Reply:
I used a for loop to truncate those sublists, however I am not sure about the tests that I used. `test_plot_dist_comparison(models, kwargs)` and `test_plot_dist_comparison_different_vars()`
Any suggestions?

Reply:
Okay so I change the code, summing up the length of each list in one variable and then compare it with `len_plots`.
I also used a for loop to truncate the sublists. However, I am not sure about the tests.

Reply:
Yeah, this is confusing since `len_plots` is different from `rcParams["plot.max_subplots"]`, and it isn't obvious to me that 
1. you can plot exactly that number of subplots
2. you won't plot more

I think 
```python
total_plots = sum(sum(1 for _ in xarray_var_iter(data, var_names=var, combined=True)) for data, var in zip(datasets, var_names))
```
gives you the number of "desired plots". I would check if that number is bigger than `rcParams["plot.max_subplots"]`, and if so, issue a warning, and say you'll plot however many things are in `dc_plotters` (which i guess is `sum(len(j) for j in dc_plotters)`). The calculation on line 134 already verifies that you shouldn't have to truncate further (I think?), so you'd just have to issue a warning.

Reply:
Specifically, I think it is an accident that the test passes, and line 137 isn't the condition we care about at all!

Reply:
Yes I also thought that at line 134 ` dc_plotters = [ list(xarray_var_iter(data, var_names=var, combined=True))[:len_plots]`dc_plotters is already truncated, so I think that it should stay as is. 
However, do you think I should compare `total_plots` with `rcParams["plot.max_subplots"]` and not with `len_plots` which divides `rcParams["plot.max_subplots"]` with `(len(groups) + 1)`, because it was [mentioned](https://github.com/arviz-devs/arviz/issues/1680#issuecomment-832994459) that we want to cut `dc_plotters` to have at most `len_plots`?